catalog/plugin/catalog-plugin-metacardbackup/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/MetacardBackupPlugin.java

```diff
@@ -194,39 +209,22 @@ private void processRequest(ProcessRequest<? extends ProcessResourceItem> proces
         }
     }
 
-    private void copyBackupToOutputDirectory(BinaryContent content, String metacardId)
+    private void backupData(BinaryContent content, String metacardId)
             throws PluginExecutionException {
         byte[] contentBytes = getContentBytes(content, metacardId);
 
-        Path metacardPath = getMetacardDirectory(metacardId);
-        if (metacardPath == null) {
-            throw new PluginExecutionException(String.format(
-                    "Unable to create metacard path directory for %s",
-                    metacardId));
-        }
+        LOGGER.trace("Writing backup from {} to backup provider", metacardId);
 
-        try {
-            Path parent = metacardPath.getParent();
-            if (parent != null) {
-                Files.createDirectories(parent);
+        for (MetacardBackupStorageProvider storageProvider : storageBackupPlugins) {
+            if (metacardOutputProviderIds.contains(storageProvider.getId())) {
+                try {
+                    storageProvider.storeData(metacardId, contentBytes);
+                } catch (IOException | MetacardBackupException e) {
+                    LOGGER.warn("Unable to backup {} to backup provider.", metacardId, e);
```

Comment:
This may be better at the debug level; with large sets of metacards i could see this polluting the logs